### ISSUE TITLE
ember-cli won't build with ember-cli version 0.2.0-beta.1

### ISSUE THREAD

**Issue by robertleib** [artificialio/ember-cli-foundation-sass#22]
```
$ ember build
version: 0.2.0-beta.1
Build failed.
/Users/robertleib/Sites/cli-test/tmp/tree_merger-tmp_dest_dir-qnD7ZpYL.tmp/app/styles/settings:58: file to import not found or unreadable: "foundation/functions"
Current dir: /Users/robertleib/Sites/cli-test/tmp/tree_merger-tmp_dest_dir-qnD7ZpYL.tmp/app/styles/
 [string exception]
Error: /Users/robertleib/Sites/cli-test/tmp/tree_merger-tmp_dest_dir-qnD7ZpYL.tmp/app/styles/settings:58: file to import not found or unreadable: "foundation/functions"
Current dir: /Users/robertleib/Sites/cli-test/tmp/tree_merger-tmp_dest_dir-qnD7ZpYL.tmp/app/styles/
 [string exception]
    at /Users/robertleib/Sites/cli-test/node_modules/ember-cli/node_modules/broccoli/lib/builder.js:35:15
    at $$$internal$$tryCatch (/Users/robertleib/Sites/cli-test/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:490:16)
    at $$$internal$$invokeCallback (/Users/robertleib/Sites/cli-test/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:502:17)
    at $$$internal$$publish (/Users/robertleib/Sites/cli-test/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:473:11)
    at $$$internal$$publishRejection (/Users/robertleib/Sites/cli-test/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:416:7)
    at Object.$$rsvp$asap$$flush [as _onImmediate] (/Users/robertleib/Sites/cli-test/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:1581:9)
    at processImmediate [as _immediateCallback] (timers.js:345:15)
```

package.json
```
{
  "name": "cli-test",
  "version": "0.0.0",
  "description": "Small description for cli-test goes here",
  "private": true,
  "directories": {
    "doc": "doc",
    "test": "tests"
  },
  "scripts": {
    "start": "ember server",
    "build": "ember build",
    "test": "ember test"
  },
  "repository": "",
  "engines": {
    "node": ">= 0.10.0"
  },
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "broccoli-asset-rev": "^2.0.0",
    "broccoli-clean-css": "^0.3.0",
    "ember-cli": "0.2.0-beta.1",
    "ember-cli-app-version": "0.3.1",
    "ember-cli-babel": "^4.0.0",
    "ember-cli-content-security-policy": "0.3.0",
    "ember-cli-dependency-checker": "0.0.7",
    "ember-cli-foundation-sass": "^1.0.0",
    "ember-cli-htmlbars": "0.7.4",
    "ember-cli-ic-ajax": "0.1.1",
    "ember-cli-inject-live-reload": "^1.3.0",
    "ember-cli-qunit": "0.3.8",
    "ember-cli-sass": "^3.0.6",
    "ember-cli-uglify": "1.0.1",
    "ember-data": "1.0.0-beta.15",
    "ember-export-application-global": "^1.0.2",
    "express": "^4.8.5",
    "glob": "^4.0.5"
  }
}

```

bower.json
```
{
  "name": "cli-test",
  "dependencies": {
    "jquery": "^1.11.1",
    "ember": "1.10.0",
    "ember-data": "1.0.0-beta.15",
    "ember-resolver": "~0.1.11",
    "loader.js": "ember-cli/loader.js#1.0.1",
    "ember-cli-shims": "ember-cli/ember-cli-shims#0.0.3",
    "ember-cli-test-loader": "ember-cli-test-loader#0.1.3",
    "ember-load-initializers": "ember-cli/ember-load-initializers#0.0.2",
    "ember-qunit": "0.2.8",
    "ember-qunit-notifications": "0.0.7",
    "qunit": "~1.17.1",
    "foundation": "~5.5.0"
  }
}
```

Brocfile.js
```
/* global require, module */

var EmberApp = require('ember-cli/lib/broccoli/ember-app');

var app = new EmberApp();

// Use `app.import` to add additional libraries to the generated
// output files.
//
// If you need to use different assets in different
// environments, specify an object as the first parameter. That
// object's keys should be the environment name and the values
// should be the asset to use in that environment.
//
// If the library that you are including contains AMD or ES6
// modules that you would like to import into your application
// please specify an object with the list of modules as keys
// along with the exports of each module as its value.

module.exports = app.toTree();
```

**Comment by Globegitter** [comment]
Ah thanks for th  detailed report. I know that ember-cli-sass had a bug in regards to that. Could you try the latest beta? 3.1.0-beta.5?

**Comment by Titinux** [comment]
Same problem with the 3.1.0-beta.5 version of ember-cli-sass but the message if shorter

```
File: /home/titi/work/SoRSS/SoRSS-cli/tmp/tree_merger-tmp_dest_dir-qgYOLGeu.tmp/app/styles/_settings.scss (58)
file to import not found or unreadable: foundation/functions
```

**Comment by Globegitter** [comment]
Ok just tested and can confirm it is not working. See https://github.com/aexmachina/ember-cli-sass#usage @robertleib or @Titinux anyone got time for a PR? Unfortunately quite busy right now and playing around with https://github.com/zurb/foundation-apps for my latest project.

**Comment by Globegitter** [comment]
I don't really mind if we break compatibility with old ember-clis

**Comment by Titinux** [comment]
I probably found the problem. Prior to ember-cli 0.2.0 the sassOptions hash is located in Ember.App (Brocfile.js) but now the sassOptions should be in the ENV hash. https://github.com/aexmachina/ember-cli-sass#ember-cli-versions-prior-to-020

config/environment.js

``` js
var ENV = {
   ...
    sassOptions: {
      includePaths: [
      'bower_components/foundation/scss'
      ]
    }
```

I will try to submit pull request as soon as possible.

**Comment by Titinux** [comment]
What do you think about #24 ?

**Comment by Globegitter** [comment]
@Titinux Thank I really appreciate the PR. Just looked at it and not sure if that will work like that. I think you need to add what it says here https://github.com/ember-cli/ember-cli/blob/master/ADDON_HOOKS.md#config into this file: https://github.com/artificialio/ember-cli-foundation-sass/blob/master/blueprints/ember-cli-foundation-sass/index.js

So add a new `config` function there. Unfortunately I also don't have very much time right now to try things out myself.

**Comment by Titinux** [comment]
Of course I tested it and it works :bowtie: 

The default implementation of the config method try to read config/environment.js file. Try to create a new ember-cli addon and you'll see that the config/environment.js file is present by default.

My first attempt was to return a hash from the method but if the hash is present in addon and in the main application, the main application win and addon config is lost. So I decided to modify the main configuration set through appConfig variable.

I also keep the old settings in the index.js file to keep compatibility. So sassOptions is defined in Ember.app for old version and in the ENV hash for new ones.
